### PR TITLE
Add updatecli manifest to update updatecli

### PR DESCRIPTION
**Pull request by timja**
I renamed the updatecli github action so that the json schema doesn't think it's an updatecli file as it looks for files called `updatecli.yaml`

### CODE REVIEW

File: updatecli/updatecli.d/updatecli.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    spec:
+      owner: "updatecli"
+      repository: "updatecli"
+      token: '{{ or .github.token (requiredEnv "UPDATECLI_GITHUB_TOKEN") }}'
+      username: "{{ .github.username }}"
+      versionfilter:
+        kind: semver
+
+conditions:
+  checkIfUpdateCliBinaryIsPublished:
```
Comment (JayDoubleu) [inline updatecli/updatecli.d/updatecli.yaml:29]: Is there a need for this ? I'm looking at https://github.com/updatecli/updatecli/blob/main/updatecli/updatecli.d/updatecli.yaml and doesnt seem to need these extra shell providers

File: .github/workflows/run-updatecli.yaml [hunk 2]
```diff
@@ -29,7 +29,13 @@ jobs:
       - name: Install Updatecli in the runner
         uses: updatecli/updatecli-action@v2
         with:
-          version: v0.33.3
+          version: v0.35.1
+
+      - uses: tibdex/github-app-token@v1
```
Comment (JayDoubleu) [inline .github/workflows/run-updatecli.yaml:34]: I'm sure it's fine but should we be looking at either locking it down to specific sha commit or forking ?

File: updatecli/updatecli.d/updatecli.yaml (hunk 1 above)
Comment (timja) [reply updatecli/updatecli.d/updatecli.yaml:29]: This came from https://github.com/jenkins-infra/jenkins-infra/blob/production/updatecli/weekly.d/updatecli.yaml#L28-L34

It's an extra safety which makes sure that we don't merge a PR before release artifacts are in place.

May be fine for this one but I've seen places take awhile to upload release artifacts even after tagging / creating a release

no harm in keeping I think

File: .github/workflows/run-updatecli.yaml (hunk 2 above)
Comment (timja) [reply .github/workflows/run-updatecli.yaml:34]: there's no policy on that here, forking is against our policy in general as we don't want to be maintaining things outside of our core business.

we could use a fixed sha

**Review by endakelly — APPROVED** [review]

**Pull request merged**